linuxkit/linuxkit
# Actions

**- What I did**

Per the discussion in #3437, having a working CI system is important to attract contributors to the project and reduce the burden on maintainers. I have spent some time looking at consolidating our CI system sprawl (CircleCI, LinuxKitCI etc..) and replacing it with GitHub Actions.

This PR introduced a single worflow for CI.
It:
- Builds binaries and runs unit tests on all supported platforms
- Runs all integration tests (in parallel)

In future this can be extended to include anything and everything that is on the CI wishlist.

A few points of note:

1. Tests are run using containerized qemu (no nested virt)
2. Overall CI time is still 1hr, but that is mainly spent running the packages tests
3. The GCP test that was running on LinuxKit CI has been ported to RTF but isn't running just yet due to security concerns around secrets
4. I'm not sure what the security model is around changes to workflow/actions. I'll clarify before merge.

**- How I did it**

Wrote lots of YAML and tested on my LinuxKit fork.

**- How to verify it**
You can see the latest green run on my fork: https://github.com/dave-tucker/linuxkit/commit/71a95bff490e7d4505767dbf337b769c58e46f16/checks?check_suite_id=329481528

**- Description for the changelog**



**- A picture of a cute animal (not mandatory but encouraged)**
![image](https://user-images.githubusercontent.com/3781777/69637766-396c0f80-1051-11ea-846a-83d602d2d881.png)
